autorest/codegen/serializers/__init__.py

```diff
@@ -491,3 +507,158 @@ def _serialize_and_write_metadata(
+            "-", "."
+        ) or self.code_model.namespace
+        imports.add_submodule_import(
+            namespace, self.code_model.client.name, ImportType.THIRDPARTY
+        )
+        if credential:
+            imports.add_import("os", ImportType.STDLIB)
+            imports.add_submodule_import(
+                "azure.identity", credential, ImportType.THIRDPARTY
+            )
+
+        return {
+            "imports": imports,
+            "client_params": client_params,
+            "auth_type": auth_type,
+        }
+
+    @staticmethod
+    def _operation_additional(operation: OperationBase[Any]) -> str:
+        lro = ".result()"
+        paging = "\n    response = [item for item in response]"
+        if operation.operation_type == "paging":
+            return "\n    response = [item for item in response]"
+        if operation.operation_type == "lro":
+            return ".result()"
+        if operation.operation_type == "lropaging":
+            return lro + paging
+        return ""
+
+    @staticmethod
+    def _to_lower_camel_case(name: str) -> str:
+        return re.sub(r"_([a-z])", lambda x: x.group(1).upper(), name)
+
+    @staticmethod
+    def _to_snake_case(name: str) -> str:
+        return re.sub(
+            "((?!^)(?<!_)[A-Z][a-z]+|(?<=[a-z0-9])[A-Z])",
+            r"_\1",
+            name.replace("-", "").replace(" ", "_"),
+        ).lower()
+
+    def _serialize_and_write_sample_folder(
+        self, env: Environment, namespace_path: Path
+    ) -> None:
+        out_path = self._package_root_folder(namespace_path) / Path("generated_samples")
+        sample_params = self._prepare_sample_render_param()
+        cls = lambda x: f'"{x}"' if isinstance(x, str) else str(x)
+        failure_info = '"fail to find required param named {%s} in example file {%s}"'
+        # pylint: disable=too-many-nested-blocks
+        for operation_group in self.code_model.operation_groups:
+            if self.code_model.options["multiapi"]:
+                api_version_folder = f"{operation_group.api_versions[0]}/"
+                sample_params["client_params"][
+                    "api_version"
+                ] = f'"{operation_group.api_versions[0]}"'
+            else:
+                api_version_folder = ""
+            for operation in operation_group.operations:
+                samples = operation.yaml_data["samples"]
+                if not samples:
+                    continue
+                params_positional = [
+                    p
+                    for p in operation.parameters.positional
+                    if not p.client_default_value
+                ]
+                operation_result = self._operation_additional(operation)
+                for key, value in samples.items():
+                    # update client parameters if it is defined in example files
+                    for param in sample_params["client_params"].keys():
+                        param_value = value["parameters"].get(
```

Comment:
we should just be able to pas in the parameters to the jinja template, and use their `client_name` properties here